ddcz/models/used/discussion.py

```diff
@@ -26,3 +28,25 @@ def user_profile_url(self):
     @property
     def by_registered_user(self):
         return self.reg == '1' and self.user
+
+    
+    def createNewCommentFromForm(self, form, user, reputation=0):
```

Comment:
I think it's better to either:

* Either have a `ModelForm` and do `form.save()`
* Have this in the view and use `Phorum.objects.create()`

But here you are leaking view tools to model. `form` should never leave the view layer—pass on the text data if needed. 

But I really see no benefit from having this as a method as opposed to creating the object in the view.

Reply:
I don't know, is it wise to work with database saving in the view? I'm somehow more comfortable to work with database only in the model. Things like that in my experience make it unreadable. So the biggest benefit is that the code in controller is readable even if it must do a lot of work. 

I was thinking of the inheriting ModelForm, but in the end, I didn't find the difference between model.save() and modelform.save() methods. I can rewrite it of course, not a problem. Just feel that I need to learn more about it so I'd understand the advantage here.

Reply:
Well, you are just pushing the unreadability into the model since suddenly, the model is not independent of the view layer (which it should). And I mean, in those case, would it be wise to *query* models in the view instead of writing a method per every query? IMHO it makes no sense, you'd encounter self-referencing cycle fairly soon. 

I think it's really best to treat models like representing "Database Records" and view as a layer to manipulate them. 

But otherwise, it's a common pattern. [See how saving is done in Django tutorial](https://docs.djangoproject.com/en/3.1/intro/tutorial04/#write-a-minimal-form) or some other Django projects.

Reply:
ad model forms, I don't have a strong opinion; I find them easier to work with when there are no modifications, but ¯\_(ツ)_/¯

Reply:
...but I still really think this should be in the view